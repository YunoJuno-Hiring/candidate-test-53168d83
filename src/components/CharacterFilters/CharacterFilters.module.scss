@import 'styles/_globals';
@import 'styles/_mixins';

.CharacterFilters-container {
  background-color: white;
  padding: 16px;
  margin: 16px;
  display: flex;
  color: $fontColor;
  flex-wrap: wrap;
  justify-content: center;
  @include lg {
    justify-content: start;
  }
}

.CharacterFilters-filter {
  margin: 0px 8px;
}
