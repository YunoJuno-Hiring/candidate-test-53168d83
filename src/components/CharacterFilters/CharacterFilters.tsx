import React from 'react';
import styles from './CharacterFilters.module.scss';
import DropdownSelector from './DropdownSelector/DropdownSelector';
import { characterCategoryStrings, orderByStrings } from '@constants/strings';

export default function CharacterFilters() {
  const characterCategoryArray: string[] = Object.values(characterCategoryStrings);
  const characterOrderByArray: string[] = Object.values(orderByStrings);

  return (
    <div className={styles['CharacterFilters-container']}>
      <div className={styles['CharacterFilters-filter']}>
        <DropdownSelector
          label="Category"
          selectOptions={['all', ...characterCategoryArray]}
          contextValue="category"
        />
      </div>
      <div className={styles['CharacterFilters-filter']}>
        <DropdownSelector
          label="Order by"
          selectOptions={characterOrderByArray}
          contextValue="orderBy"
        />
      </div>
    </div>
  );
}
