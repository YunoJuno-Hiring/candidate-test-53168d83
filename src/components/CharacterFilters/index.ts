export { default as CharacterFilters } from './CharacterFilters';
