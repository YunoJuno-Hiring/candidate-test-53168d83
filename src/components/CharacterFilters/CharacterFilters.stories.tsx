import React from 'react';

import type { Meta, StoryObj } from '@storybook/react';
import CharacterFilters from './CharacterFilters';
import FilterContextProvider from '@contexts/FilterContext';

const meta: Meta<typeof CharacterFilters> = {
  component: CharacterFilters,
  decorators: [
    (Story) => (
      <FilterContextProvider>
        <Story />
      </FilterContextProvider>
    )
  ]
};

export default meta;

type Story = StoryObj<typeof CharacterFilters>;

export const Default: Story = {
  render: () => <CharacterFilters />
};
