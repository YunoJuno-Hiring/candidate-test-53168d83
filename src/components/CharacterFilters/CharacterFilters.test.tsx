import FilterContextProvider from '@contexts/FilterContext';
import React from 'react';
import renderer from 'react-test-renderer';
import CharacterFilters from './CharacterFilters';

describe('CharacterFilters', () => {
  it('Should match the DOM snapshot', () => {
    const component = renderer
      .create(
        <FilterContextProvider>
          <CharacterFilters />
        </FilterContextProvider>
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
