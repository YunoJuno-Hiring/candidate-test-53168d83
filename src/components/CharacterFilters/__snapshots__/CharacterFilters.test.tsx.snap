// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterFilters Should match the DOM snapshot 1`] = `
<div
  className="CharacterFilters-container"
>
  <div
    className="CharacterFilters-filter"
  >
    <div>
      <p
        className="DropdownSelector-label"
      >
        Category
      </p>
      <select
        className="DropdownSelector"
        onChange={[Function]}
        value="all"
      >
        <option
          value="all"
        >
          All
        </option>
        <option
          value="corrupted hobbit"
        >
          Corrupted Hobbit
        </option>
        <option
          value="dark lord"
        >
          Dark Lord
        </option>
        <option
          value="dwarf"
        >
          Dwarf
        </option>
        <option
          value="elf"
        >
          Elf
        </option>
        <option
          value="ent"
        >
          Ent
        </option>
        <option
          value="hobbit"
        >
          Hobbit
        </option>
        <option
          value="horse"
        >
          Horse
        </option>
        <option
          value="human"
        >
          Human
        </option>
        <option
          value="spider"
        >
          Spider
        </option>
        <option
          value="spirit"
        >
          Spirit
        </option>
        <option
          value="wizard"
        >
          Wizard
        </option>
      </select>
    </div>
  </div>
  <div
    className="CharacterFilters-filter"
  >
    <div>
      <p
        className="DropdownSelector-label"
      >
        Order by
      </p>
      <select
        className="DropdownSelector"
        onChange={[Function]}
        value="alphabetical"
      >
        <option
          value="alphabetical"
        >
          Alphabetical
        </option>
        <option
          value="significance"
        >
          Significance
        </option>
      </select>
    </div>
  </div>
</div>
`;
